lib/koans.ex

```diff
@@ -82,11 +82,32 @@ defmodule Koans do
   end
 
   defmacro __before_compile__(env) do
-    koans = Module.get_attribute(env.module, :koans) |> Enum.reverse
+    koans = koans(env)
+    intro = extract_into(env)
     quote do
       def all_koans do
         unquote(koans)
       end
+
+      def intro do
+        unquote(intro)
```

Comment:
If every module has `@intro` then you are right.
I could flick this around and make sure that `use Koans` adds the attribute with "" if there is nothing defined in the module itself.

What do you think?